src/gui/MenuBar.cpp

```diff
@@ -466,6 +494,89 @@ void MenuBar::setKeyShortcuts()
+	QSettings profileSettings(QMPlay2Core.getSettingsDir() + "Profile.ini", QSettings::IniFormat);
+	if (selectedProfile != profileSettings.value("Profile", "/").toString())
+	{
+		profileSettings.setValue("Profile", selectedProfile);
+		QMPlay2GUI.restartApp = true;
+		QMPlay2GUI.mainW->close();
+	}
+}
+void MenuBar::addProfile()
+{
+	const QString selectedProfile = Functions::cleanFileName(QInputDialog::getText(
+				this, Options::tr("Create new profile"),
+				Options::tr("Profile name")));
+	if (selectedProfile.isEmpty())
+		return;
+	QSettings profileSettings(QMPlay2Core.getSettingsDir() + "Profile.ini", QSettings::IniFormat);
+	if (selectedProfile != profileSettings.value("Profile", "/").toString())
+	{
+		QDir(QMPlay2Core.getSettingsDir()).mkpath("Profiles/" + selectedProfile);
+		QFile tempFile(QMPlay2Core.getSettingsDir() + "Profiles/" + selectedProfile + "/QMPlay2.ini");
+		tempFile.open(QFile::WriteOnly);
+		tempFile.close();
+
+		profileSettings.setValue("Profile", selectedProfile);
+		QMPlay2GUI.restartApp = true;
+		QMPlay2GUI.mainW->close();
+	}
+}
+void MenuBar::copyProfile()
+{
+	const QString selectedProfile = Functions::cleanFileName(QInputDialog::getText(
+				this, Options::tr("Create new profile"),
+				Options::tr("Profile name")));
+	if (selectedProfile.isEmpty())
+		return;
+	QSettings profileSettings(QMPlay2Core.getSettingsDir() + "Profile.ini", QSettings::IniFormat);
+	if (selectedProfile != profileSettings.value("Profile", "/").toString())
+	{
+		QDir(QMPlay2Core.getSettingsDir()).mkpath("Profiles/" + selectedProfile);
+		const QString srcDir = QMPlay2Core.getSettingsDir() + QMPlay2Core.getSettingsProfile() + "/";
+		const QString dstDir = QMPlay2Core.getSettingsDir() + "Profiles/" + selectedProfile + "/";
+		foreach (const QString &path, QDir(srcDir).entryList(QStringList() << "*.ini", QDir::Files))
+		{
+			QFile::copy(srcDir + path, dstDir + path);
+		}
+
+		profileSettings.setValue("Profile", selectedProfile);
+		QMPlay2GUI.restartApp = true;
+		QMPlay2GUI.mainW->close();
+	}
+}
+void MenuBar::removeProfileMenuRequest(const QPoint &p)
+{
+	QMenu *menu = qobject_cast<QMenu *>(sender());
+	QAction *act = menu->actionAt(p);
+	if (act && menu->actions().indexOf(act) >= 4)
+	{
+		options->removeProfileMenu->setProperty("profile", QVariant::fromValue((void *)act));
+		options->removeProfileMenu->popup(menu->mapToGlobal(p));
+	}
+}
+void MenuBar::removeProfile()
+{
+	if (QAction *act = (QAction *)options->removeProfileMenu->property("profile").value<void *>())
+	{
+		const QString &profile = act->property("path").toString();
+		Functions::dirRemoveRecursively(QMPlay2Core.getSettingsDir() + "Profiles/" + profile);
+		if("Profiles/" + profile + "/" == QMPlay2Core.getSettingsProfile())
+		{
+			QSettings profileSettings(QMPlay2Core.getSettingsDir() + "Profile.ini", QSettings::IniFormat);
+			profileSettings.setValue("Profile", "/");
+			QMPlay2GUI.restartApp = QMPlay2GUI.removeSettings = true;
```

Comment:
`QMPlay2GUI.removeSettings` also removes settings, so is the `Functions::dirRemoveRecursively` necessary? maybe simple `rmdir` would be enough after profile removal?